examples/prom/de_parametric_maxwell_greedy.cpp

```diff
@@ -0,0 +1,943 @@
+                   "The number of values to search in the parameter point space.");
+    args.AddOption(&greedy_relative_error_tol, "-greedyrelerrortol",
+                   "--greedyrelerrortol", "The greedy algorithm relative error tolerance.");
+    args.AddOption(&greedy_subset_size, "-greedysubsize", "--greedysubsize",
+                   "The greedy algorithm subset size.");
+    args.AddOption(&greedy_convergence_subset_size, "-greedyconvsize",
+                   "--greedyconvsize", "The greedy algorithm convergence subset size.");
+    args.AddOption(&de_min_freq, "-de_min_freq", "--de_min_frequency",
+                   "DE min freq.");
+    args.AddOption(&de_max_freq, "-de_max_freq", "--de_max_frequency",
+                   "DE max freq.");
+    args.AddOption(&de_F, "-de_f", "--de_f",
+                   "DE F.");
+    args.AddOption(&de_CR, "-de_cr", "--de_cr",
+                   "DE CR.");
+    args.AddOption(&de_PS, "-de_ps", "--de_ps",
+                   "DE Population size.");
+    args.AddOption(&de_min_iter, "-de_min_iter", "--de_min_iter",
+                   "DE Minimum number of iterations.");
+    args.AddOption(&de_max_iter, "-de_max_iter", "--de_max_iter",
+                   "DE Maximum number of iterations.");
+    args.AddOption(&de_ct, "-de_ct", "--de_ct",
+                   "DE Cost threshold.");
+    args.AddOption(&de, "-de", "--de", "-no-de", "--no-de",
+                   "Enable or disable the differential evolution phase.");
+
+    args.Parse();
+    if (!args.Good())
+    {
+        if (myid == 0)
+        {
+            args.PrintUsage(cout);
+        }
+        return 1;
+    }
+    if (myid == 0)
+    {
+        args.PrintOptions(cout);
+    }
+
+    MFEM_VERIFY(!(build_database
+                  && de), "both build_database and de can not be true!");
+
+    MFEM_VERIFY(!build_database
+                || !use_database,
+                "both build_database and use_database can not be used at the same time.");
+
+    if (de)
+    {
+        calc_err_indicator = false;
+        calc_rel_error = false;
+        online = true;
+        build_database = false;
+        use_database = true;
+    }
+
+    // 3a. Set up the ROM database for the greedy algorithm to run.
+    if (build_database)
+    {
+        MFEM_VERIFY(!offline
+                    && !online,
+                    "offline and online must be turned off during the build_database phase.");
+        MFEM_VERIFY(!visit
+                    && !visualization,
+                    "visit and visualization must be turned off during the build_database phase.")
+        std::ifstream infile("de_parametric_maxwell_greedy_algorithm_data");
+        if (infile.good())
+        {
+            if (myid == 0) cout << "The database has already been built."
+                                    << "Exiting." << endl;
+            return 0;
+        }
+        infile.close();
+        greedy_sampler = new CAROM::GreedyRandomSampler(greedy_param_space_min,
```

Comment:
`greedy_sampler` is not deleted.

Reply:
I delete it now after building the database. It was also being initialized twice, here and on line 775. I deleted the second instance of initialization.